manticore/platforms/linux.py

```diff
@@ -986,20 +1000,18 @@ def sys_lseek(self, fd, offset, whence):
 
         :return: 0 (Success), or EBADF (fd is not a valid file descriptor or is not open)
 
-         '''
-        if not self._is_open(fd):
-            logger.info("LSEEK: Not valid file descriptor on lseek. Returning EBADF")
-            return -errno.EBADF
-
-        if isinstance(self.files[fd], Socket):
-            logger.info("LSEEK: Not valid file descriptor on lseek. Fd not seekable. Returning EBADF")
+        '''
+        try:
+            # Read the data and put it in memory
+            if not self._is_open(fd):
+                raise BadFd()
+            self.files[fd].seek(offset)
+        except BadFd:
```

Comment:
Yep I don't like this handling logic either.. this was there just since both a seek()
on Socket and a non-open fd both return a EBADF. Essentially we would like to ignore what type of fd we have and let the classes handle this, this is why I wanted to remove all the isinstance() cases in handling logic. Unfortunately, for _is_open, self.files can have an idx with a None value for a closed fd, so IndexError will require us tweaking the self.files logic. What do you think?